WalletWasabi/Extensions/LinqExtensions.cs

```diff
@@ -215,5 +216,11 @@ public static IEnumerable<(T1, T2)> ZipForceEqualLength<T1, T2>(this IEnumerable
 
 		public static bool IsSuperSetOf<T>(this IEnumerable<T> me, IEnumerable<T> other) =>
 			other.All(x => me.Contains(x));
+
+		public async static Task<TResult> Then<TResult>(this Task source, Task<TResult> continuation)
```

Comment:
```suggestion
		public async static Task<TResult> ThenAsync<TResult>(this Task source, Task<TResult> continuation)
```

Reply:
I prefer `Then`. Microsoft suggests the `Async` naming convention for us but they don't use it for them because that could ruin their APIs, for example, the named it `Task.Delay()` instead of `Task.DelayAsync()`.

Reply:
> Microsoft suggests the `Async` naming convention for us but they don't use it for them because that could ruin their APIs, for example, the named it `Task.Delay()` instead of `Task.DelayAsync()`.

The argument does not make sense to me: Yes, Microsoft named it `Task.Delay()` instead of `Task.DelayAsync()` but like 99.9999% of the new code follows the `Async` suffix convention. So I consider `Task.Delay()` to be a historical mistake (though this just a guess on my side).

Also we have in editorconfig a rule that warns us when `Async` suffix is missing so you introduce a new warning for no good reason. If the reason is good, then the warning should be at least suppressed.

Reply:
It is not an historial mistake, the new methods also avoid the `Async` as `Task.WhenAll` or `ConfigureAwait`, they have clear names, otherwise the code would look like `var results = await Task.WhenAllAsync(TaskDelayAsync(1).ConfigureAwaitAsync(false))`.

Your second point is more convincing.

Reply:
This is an "official" explanation: https://github.com/dotnet/runtime/discussions/59761. FTR :-)

Reply:
since it's not a Linq extension, does it make sense to put it in e.g. `TaskExtensions`, and then disable the warning in that file based on the same reasoning that the async is implied by the class name?

Reply:
I think the most simple solution would be *not having* the `ThenAsync` method at all. Extension methods are supposed to be very useful and used by many. `ThenAsync` method just puts two commands one after another.

Reply:
done.